arches/app/models/models.py

```diff
@@ -2190,3 +2180,348 @@ def __str__(self):
+    controlled_list = models.ForeignKey(
+        ControlledList,
+        db_column="listid",
+        on_delete=models.CASCADE,
+        related_name="controlled_list_items",
+    )
+    sortorder = models.IntegerField(validators=[MinValueValidator(0)])
+    parent = models.ForeignKey(
+        "self", null=True, blank=True, on_delete=models.CASCADE, related_name="children"
+    )
+    guide = models.BooleanField(default=False)
+
+    class Meta:
+        db_table = "controlled_list_items"
+        constraints = [
+            # Sort order concerns the list as a whole, not subsets
+            # of the hierarchy.
+            models.UniqueConstraint(
+                fields=["controlled_list", "sortorder"],
+                name="unique_list_sortorder",
+                deferrable=Deferrable.DEFERRED,
+                violation_error_message=_(
+                    "All items in this list must have distinct sort orders."
+                ),
+            ),
+            models.UniqueConstraint(
+                fields=["controlled_list", "uri"],
+                name="unique_list_uri",
+                violation_error_message=_(
+                    "All items in this list must have distinct URIs."
+                ),
+            ),
+        ]
+
+    def clean(self):
+        if not self.controlled_list_item_values.filter(valuetype="prefLabel").exists():
+            raise ValidationError(_("At least one preferred label is required."))
+
+    def clean_fields(self, exclude=None):
+        super().clean_fields(exclude=exclude)
+        if (not exclude or "uri" not in exclude) and not self.uri:
+            self.uri = None
+
+    def serialize(self, depth_map=None, flat=False):
+        if depth_map is None:
+            depth_map = defaultdict(int)
+        if self.parent_id:
+            depth_map[self.id] = depth_map[self.parent_id] + 1
+        data = {
+            "id": str(self.id),
+            "controlled_list_id": str(self.controlled_list_id),
+            "uri": self.uri,
+            "sortorder": self.sortorder,
+            "guide": self.guide,
+            "values": [
+                value.serialize()
+                for value in self.controlled_list_item_values.all()
+                if value.valuetype_id != "image"
+            ],
+            "images": [
+                image.serialize() for image in self.controlled_list_item_images.all()
+            ],
+            "parent_id": str(self.parent_id) if self.parent_id else None,
+            "depth": depth_map[self.id],
+        }
+        if not flat:
+            data["children"] = sorted(
+                [child.serialize(depth_map, flat) for child in self.children.all()],
+                key=lambda d: d["sortorder"],
+            )
+        return data
+
+
+class ControlledListItemValue(models.Model):
+    id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
+    list_item = models.ForeignKey(
```

Comment:
This last-minute renaming anticipates a little subset of the renaming of #11072 just to achieve the benefit of bringing the frontend and backend names into slightly closer alignment (and allow the view to just `Model(**data)`).